Jenkinsfile

```diff
@@ -8,9 +8,23 @@ elifePipeline {
     node('containers-jenkins-plugin') {
         stage 'Build images', {
             checkout scm
+            sh "find build/critical-css -name '*.css' -type f -delete"
+            sh "touch build/critical-css/default.css"
             dockerComposeBuild commit
         }
 
+        stage 'Generate critical CSS', {
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml up -d critical_css"
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml exec -T critical_css node_modules/.bin/gulp critical-css:generate"
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml exec -T critical_css ./check_critical_css.sh"
+            sh "docker cp journal_critical_css_1:build/critical-css/. build/critical-css/"
+            sh "docker-compose -f docker-compose.yml -f docker-compose.ci.yml down -v"
+        }
+
+        stage 'Rebuild FPM image', {
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml build fpm"
```

Comment:
Image not immutable, but not pushed; no problems with the cache, but tricky to reason about. Wonder if we can use different tags, but can revisit later.

Reply:
e.g. pass a `IMAGE_TAG_SUFFIX` variable that is only used for `fpm` and that is initially set as `-incomplete` (or similar value) while it is empty `` on rebuild

Reply:
Alternative: can the folder be published as an independent container and volumes being used to link `fpm` and `critical-css`?

Reply:
Trying to avoid having 3 published containers (we already have `fpm` and `assets` - to be served by Nginx).